### DISCUSSION TITLE
Recommended number of passes to run

### DISCUSSION THREAD

**Discussion by matthuszagh** [memtest86plus/memtest86plus#342] [Q&A] [answered]
Is there a recommended number of passes to run to have a fairly high level of confidence that memory/mobo/CPU are working correctly? Are subsequent passes more exhaustive or do they otherwise differ in some way from previous passes, or is the logic behind running multiple passes that the false negative rate is x^n, where x is the false negative rate for one pass?

[This link](https://www.tenforums.com/tutorials/14201-memtest86-test-ram.html) recommends >= 8 passes. [This post by a maintainer](https://github.com/memtest86plus/memtest86plus/discussions/331#discussioncomment-6688218) indicates subsequent passes are more thorough than the first pass, possibly providing a partial answer to one of my questions above.

I expect that recommending a specific number of passes might be a bit arbitrary. But it would be really useful to know what developers or frequent users do for their own machines.

Thanks!

**Comment by x86fr** [comment, marked as answer]
Let's say that the first pass of Memtest86+ achieves a 99% confidence level (ie: your memory is either good or bad), and a second pass 99.9%. The first pass has reduced iterations because in production environment, we want to find the first error as fast as possible.  
Basically, we want to know if the memory is good or not , and not how bad it is.

**Reply by matthuszagh** [reply]
Ok thanks very much for your reply. It sounds like 2 full passes should be more than adequate for most cases then. I've heard people say 8 times or more (as mentioned), or up to 48 hours (particulary when overclocking). Is there any basis to this, or is that total overkill?

**Reply by x86fr** [reply]
If you computer is overclocked or running critical applications, it might be a good idea to let it run 3,4,5 passes, or overnight. But for the majority of users, 1 or 2 passes is often enough to spot the vast majority of memory issues.

**Reply by matthuszagh** [reply]
Thank you!!
